builder-image-terraform/Dockerfile

```diff
@@ -15,14 +15,25 @@ ENV GOPATH /go
 ENV PATH /go/bin:$PATH
 RUN mkdir -p ${GOPATH}/src ${GOPATH}/bin
 
-ENV TERRAFORM_VERSION 0.12.12
-RUN curl -f -Lo terraform.zip https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
-  unzip terraform.zip && \
-  rm -f terraform.zip && \
-  chmod +x terraform && \
-  mv terraform /usr/bin
+WORKDIR /tmp/
+ENV TERRAFORM_VERSION 0.12.13
+COPY sig/hashicorp.asc /tmp/.
+ADD https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip /tmp/
+ADD https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_SHA256SUMS /tmp/
+ADD https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_SHA256SUMS.sig /tmp/
 
-ENV TERRAGRUNT_VERSION 0.20.4
+RUN apk --update add --virtual verify gpgme \
+ && gpg --import hashicorp.asc \
+ && gpg --verify /tmp/terraform_${TERRAFORM_VERSION}_SHA256SUMS.sig \
+ && cat terraform_${TERRAFORM_VERSION}_SHA256SUMS | grep linux_amd64 | sha256sum -c \
+ && unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip \
+ && chmod +x terraform \
+ && mv terraform /usr/bin \
+ && apk del verify \
```

Comment:
would it be better to do that as a multi stage dockerfile so we don't have to uninstall and clean up?

Reply:
As long as it's all in one step, it should layer thin, I thought? I can remove it or try to move it to a multi-stage build if that's preferred.